src/fabian_nemazi_ex/ex06/logistic_regression.py

```diff
@@ -0,0 +1,299 @@
+            The observed classes for each data point in X.
+        """
+        # This function ensures that X and y has acceptable data types
+        # and flattens y to have shape (n,) if it has shape (n, 1)
+        X, y = check_X_y(X, y, order="C")
+
+        if any((y < 0) | (y > 1)):
+            raise ValueError("Only y-values between 0 and 1 are accepted.")
+
+        # A random state is a random number generator, akin to those
+        # you made in earlier coursework. It has all functions of
+        # np.ranom, but its sequence of random numbers is not affected
+        # by calls to np.random.
+        random_state = check_random_state(self.random_state)
+        coef = random_state.standard_normal(X.shape[1])
+
+        self.coef_ = self._fit_gradient_descent(coef, X, y)
+        return self
+
+    def predict_proba(self, X):
+        """Estimate the class probabilities.
+        This function returns the probability that each datapoint belongs to
+        the positive class.
+        Parameters
+        ----------
+        X : np.ndarray
+            The data matrix.
+        Returns
+        -------
+        p : np.ndarray
+            A vector of probabilities. The i-th entry is the probability for
+            the i-th data point belonging to the positive class.
+        """
+        if not hasattr(self, "coef_"):
+            raise NotFittedError("Call fit before prediction")
+        return predict_proba(self.coef_, X)
+
+    def predict_log_proba(self, X):
+        """Estimate the class log probabilities.
+        This function returns the probability that each datapoint belongs to
+        the positive class.
+        Parameters
+        ----------
+        X : np.ndarray
+            The data matrix.
+        Returns
+        -------
+        lp : np.ndarray
+            A vector of log probabilities. The i-th entry is the log
+            probability for the i-th data point belonging to the positive
+            class.
+        """
+        return np.log(self.predict_proba(X))
+
+    def predict(self, X):
+        """Predict whether each data point in X belongs to the positive class
+        Parameters
+        ----------
+        X : np.ndarray
+            Data matrix
+        Returns
+        -------
+        yhat : np.ndarray
+            Predicted classes for the input data matrix. len(yhat) == len(X)
+        """
+        return self.predict_proba(X) >= 0.5
+
+
+if __name__ == "__main__":
+    # Simulate a random dataset
+    X = np.random.standard_normal((100, 5))
+    coef = np.random.standard_normal(5)
+    y = predict_proba(coef, X) > 0.5
+
+    # Fit a logistic regression model to the X and y vector
+    # Fill in your code here.
+    # Create a logistic regression object and fit it to the dataset
+
+    # Print performance information
+    # print(f"Accuracy: {lr_model.score(X, y)}")
+    print(f"True coefficients: {coef}")
+    # print(f"Learned coefficients: {lr_model.coef_}")
```

Comment:
You are missing the code to create a logistic model here.
```
lr_model = LogisticRegression(max_iter=30)
lr_model.fit(X, y)
```

Then you can uncomment line 294 and 296 to see the result.